user/libpinyin/template.py

```diff
@@ -0,0 +1,35 @@
+pkgname = "libpinyin"
+pkgver = "2.9.92"
+pkgrel = 0
+build_style = "gnu_configure"
+configure_args = [
+    "--with-dbm=KyotoCabinet", 
+    "--enable-libzhuyin"
+]
+hostmakedepends = [
+    "pkgconf", 
+    "autoconf", 
+    "automake", 
+    "libtool",
+    "wget2",
+    "gettext"
+]
+
+makedepends = [
+    "kyotocabinet-devel", 
+    "glib-devel"
+]
+    
+depends = ["pcre2-devel"]
```

Comment:
```suggestion
```

 
this would be makedepends if it was used, but it's not used at all

Reply:
just ran ldd /lib/libpinyin.so:
ldd (0x77330d06c000)
	libglib-2.0.so.0 => /lib/libglib-2.0.so.0 (0x77330cda7000)
	libkyotocabinet.so.16 => /lib/libkyotocabinet.so.16 (0x77330ccaf000)
	libc++.so.1 => /lib/libc++.so.1 (0x77330cba9000)
	libc++abi.so.1 => /lib/libc++abi.so.1 (0x77330cb67000)
	libunwind.so.1 => /lib/libunwind.so.1 (0x77330cb5b000)
	libc.so => ldd (0x77330d06c000)
	libpcre2-8.so.0 => /lib/libpcre2-8.so.0 (0x77330caa7000)
	libz.so.1 => /lib/libz.so.1 (0x77330ca7e000)

Not sure why this is linked, because we haven't seen pcre2-devel used during build stage. So I added pcre2-devel to depends for safety.

Reply:
that's not a dependency of libpinyin, and it's not linked at all, it's a dependency of glib:

```
$ readelf -d /usr/lib/libpinyin.so.15
  0x0000000000000001 (NEEDED)       Shared library: [libglib-2.0.so.0]
  0x0000000000000001 (NEEDED)       Shared library: [libkyotocabinet.so.16]
  0x0000000000000001 (NEEDED)       Shared library: [libc++.so.1]
  0x0000000000000001 (NEEDED)       Shared library: [libc++abi.so.1]
  0x0000000000000001 (NEEDED)       Shared library: [libunwind.so.1]
  0x0000000000000001 (NEEDED)       Shared library: [libc.so]

^ no pcre2 at all, it is not linked

$ readelf -d /usr/lib/libglib-2.0.so.0 
  Tag                Type           Name/Value
  0x0000000000000001 (NEEDED)       Shared library: [libpcre2-8.so.0]
  0x0000000000000001 (NEEDED)       Shared library: [libc.so]
  0x000000000000000e (SONAME)       Library soname: [libglib-2.0.so.0]
```

and if pcre2 was linked to libpinyin, cbuild would implicitly add a dependency for it already (you would see `SONAME: libpcre2-8.so.0` in the runtime_deps hook like the other deps)

this pcre2-devel doesn't do anything except add an unused and wrong dependency. `ldd` does not tell you what libraries something links to, it only tries to load an elf and then tells you every library it uses recursively in a flat way, which is not useful to see what something directly depends on.

i don't understand why i have to repeat everything to you multiple times..

Reply:
okay... sorry, not a developer myself, just a user trying to get software running on this distro. there's a lot of things for me to learn. will do more investigations next time..

Reply:
sure, you've done a good job so far :)